moshi/src/main/java/com/kroger/moshi/CacheEntryListSerializer.kt

```diff
@@ -29,16 +29,31 @@ import com.squareup.moshi.JsonReader
 import com.squareup.moshi.JsonWriter
 import javax.inject.Inject
 
+/**
+ * An implementation of [JsonAdapter] for handling a list of [CacheEntry] objects
+ *
+ * @property cacheEntrySerializer [CacheEntrySerializer] for serializing individual [CacheEntry] objects
+ */
 public class CacheEntryListSerializer<K, V> @Inject constructor(private val cacheEntrySerializer: CacheEntrySerializer<K, V>) : JsonAdapter<List<CacheEntry<K, V>>>() {
-    override fun fromJson(p0: JsonReader): List<CacheEntry<K, V>> {
-        val items = p0.nextString().split(";")
+    override fun fromJson(reader: JsonReader): List<CacheEntry<K, V>> {
+        val entries = mutableListOf<CacheEntry<K, V>>()
+
+        reader.beginArray()
+        while (reader.hasNext()) {
+            entries.add(cacheEntrySerializer.fromJson(reader))
+        }
+        reader.endArray()
 
-        return items.map { hydrateCacheEntry(it, cacheEntrySerializer.keyAdapter, cacheEntrySerializer.valueAdapter) }
+        return entries.toList()
     }
 
-    override fun toJson(p0: JsonWriter, p1: List<CacheEntry<K, V>>?) {
-        val value = p1?.joinToString(";") { flattenCacheEntry(it, cacheEntrySerializer.keyAdapter, cacheEntrySerializer.valueAdapter) }
+    override fun toJson(writer: JsonWriter, cacheEntries: List<CacheEntry<K, V>>?) {
+        require(!cacheEntries.isNullOrEmpty())
```

Comment:
Should it be an empty file or an empty array?

Reply:
If we do that it should be documented. Without being told otherwise I would expect to get the same value I serialized back when deserializing. null -> null, empty list -> empty list.

Reply:
Updated to return an empty list if that was set, null is handled by `MoshiCacheSerializer`